taichi/ir/frontend_ir.cpp

```diff
@@ -401,7 +399,30 @@ Stmt *make_field_access(Expression::FlattenContext *ctx,
     }
     index_stmts.push_back(ind);
   }
-  return ctx->push_back(std::make_unique<GlobalPtrStmt>(snode, index_stmts));
+  return index_stmts;
+}
+
+Stmt *make_field_access(Expression::FlattenContext *ctx,
+                        Expr var,
+                        ExprGroup indices) {
+  SNode *snode = var.cast<FieldExpression>()->snode;
+  return ctx->push_back(std::make_unique<GlobalPtrStmt>(
+      snode, make_index_stmts(ctx, indices, snode->index_offsets)));
+}
+
+Stmt *make_matrix_field_access(Expression::FlattenContext *ctx,
+                               Expr var,
+                               ExprGroup indices,
+                               DataType ret_type) {
+  auto matrix_field = var.cast<MatrixFieldExpression>();
```

Comment:
assert `matrix_field != nullptr` after dynamic pointer cast? Or take `MatrixFieldExpression var` explicitly?

Reply:
Sure. I've chosen the latter.